src/compiler/scala/tools/nsc/typechecker/ContextErrors.scala

```diff
@@ -46,9 +46,10 @@ trait ContextErrors {
     extends TreeTypeError
 
   /**
-   * Marks a TypeError that was constructed from a CyclicReference (under silent).
-   * This is used for named arguments, where we need to know if an assignment expression
-   * failed with a cyclic reference or some other type error.
+   * Marks a TypeError that was constructed from a CyclicReference (under silent). This was used
+   * up to 2.12 to identify cyclic references when type checking a named argument as if it were an
+   * assignment (to figure out if the expression was a valid assignment, a4a892fb01). Since 2.13 we
+   * could use a normal NormalTypeError, but I'll leave it in for now, it might come in handy.
    */
   class NormalTypeErrorFromCyclicReference(underlyingTree: Tree, errMsg: String)
```

Comment:
I'm always weary of leaving unused stuff in, because some future dev may misunderstand its purpose and wreak havoc with it :-)

Reply:
OK, I'm happy to remove it.